hibernate-rx-core/src/main/java/org/hibernate/rx/event/impl/DefaultRxLoadEventListener.java

```diff
@@ -511,7 +595,7 @@ private Object createProxy(
 	 *
 	 * @return The loaded entity, or null.
 	 */
-	private CompletionStage<Optional<Object>> doLoad(
```

Comment:
@DavideD sorry if I'm a bit late in noticing this change, but I wonder why you made it?

Ultimately I think the pattern of returning `CompletionStage`s from methods of type `Object` that usually return the entity itself is something we'll eventually want to move away from (though it's OK for now).

So it feels like the return type was more correct before you made this change.

Reply:
@DavideD And I see a similar change was made to three other methods. So I suppose you had a good reason for wanting to do that, but I wonder what it was.

Reply:
I think the problem is that sometimes ORM calls `org.hibernate.internal.SessionImpl#internalLoad` that `callsDefaultRxLoadEventListener#onLoad` instead of the reactive `rxOnLoad`.
rxOnLoad and onLoad do basically the same thing but because onLoad is not reactive I cannot handle a CompletionStage withtout waiting for its completion first. My assumption is that a CompletionStage will only be returned when the method doLoad needs to execute a query on the db. In all the other cases it returns a proxy and the call to onLoad is not a problem.

That's why I've changed it the doLoad and the other methods to return an object, this way I can then check if it's a CompletionStage or not and throw an exception if I'm in the onLoad method. Basically, it is to avoid calling the .get() method and shows a blocking point in the code.

These are all private methods so I didn't expect it to be that important. Plus, I like that the final CompletionStage is created just before returning from rxOnLoad.

Reply:
But maybe there was a simple way to do it.

Reply:
I'm workin on it now.

Reply:
@DavideD See https://github.com/hibernate/hibernate-rx/pull/88.